specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json

```diff
@@ -1930,6 +2076,100 @@
         }
       },
       "description": "The backup key result, containing the backup blob."
+    },
+    "KeyPolicy": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The key policy id.",
+          "readOnly": true
+        },
+        "lifetimeActions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LifetimeActions"
+          },
+          "description": "Actions that will be performed by Key Vault over the lifetime of a key."
+        },
+        "attributes": {
+          "$ref": "#/definitions/KeyPolicyAttributes",
+          "description": "The key policy attributes."
+        }
+      },
+      "description": "Management policy for a key."
+    },
+    "LifetimeActions": {
+      "properties": {
+        "trigger": {
+          "$ref": "#/definitions/LifetimeActionsTrigger",
+          "description": "The condition that will execute the action."
+        },
+        "action": {
+          "$ref": "#/definitions/LifetimeActionsType",
+          "description": "The action that will be executed."
+        }
+      },
+      "description": "Action and its trigger that will be performed by Key Vault over the lifetime of a key."
+    },
+    "LifetimeActionsTrigger": {
+      "properties": {
+        "timeAfterCreation": {
+          "type": "string",
+          "format": "string",
+          "minimum": 1,
+          "maximum": 99,
+          "description": "Days after creation to attempt rotate."
```

Comment:
Do we support units other than days?

Reply:
Not initially. The type is a `string` exactly to allow supporting other units in the future without breaking the API. Initially, the unit validation happens at server side only.

But we should definitely change documentation as "Days after creation to attempt rotate, **expressed in ISO 8601 duration**".

Reply:
This is going to be problematic for SDKs, especially for strongly-typed languages. Even our duck-typed languages' generators will have problems with this. Why not just be explicit and add "Days" to the property. Should you ever need to change it, you could support an alternate property. We'd document to do one or the other, but would serialize both (we don't do more client-side validation than necessary to call REST methods, avoiding tight coupling otherwise).

Reply:
If nothing else, why not just go with ISO 8601-compliant strings now and be consistent? This way, if users were having problems constructing proper strings, we could universally provide some sort of converter to convert `TimeSpan` values (or whatever a language uses) into proper ISO 8601-compliant strings like "P90D".

Reply:
@heaths 

> If nothing else, why not just go with ISO 8601-compliant strings now and be consistent? 

That's what I've meant when I said **expressed in ISO 8601 duration** in my previous comment. The thing is that we won't allow values such as `PT6H`. The server will only allow values such as `P1D` or `P180D`. The `minimum` and `maximum` properties also need to be removed from this specification.